core/src/orphan_manager.rs

```diff
@@ -85,7 +89,11 @@ impl<'gc> OrphanManager<'gc> {
                 // which have not been moved in the displaylist by ActionScript. Therefore,
                 // any orphan we see that has 'placed_by_script()' should stay on the orphan
                 // list, because it was not removed by a RemoveObject tag.
-                dobj.placed_by_script()
+                //
+                // Also, we consider AVM2 MovieClips loaded into an AVM1 parent to always
+                // be orphans, since we know they were placed by AVM1 code and not by
+                // the timeline.
+                dobj.placed_by_script() || has_avm1_parent
```

Comment:
Shouldn't such movies be placed by script anyway?

Reply:
Oh right, but in this case you have `placed_by_avm1_script`. As a side note, I think we should aim at merging these two.

Reply:
Maybe I'm misunderstanding something, but the comment says that we do that exactly because the object couldn't have been placed by the timeline. In which case it makes sense to check `placed_by_avm1_script` instead of using this assumption, right?

I'm looking into merging `placed_by_avm1_script` and `placed_by_avm2_script` into one